src/components/Button/Button.tsx

```diff
@@ -2,12 +2,15 @@ import React, { memo, forwardRef, ComponentProps } from "react"
 import { cx } from "class-variance-authority"
 import { button, interactive } from "./utils"
 import { ButtonVariants, InteractiveVariants } from "./typings"
-import { useButtons } from "../Buttons/hooks"
+import { useButtons } from "../Buttons"
+import { Colors } from "../../../drxkicui.config.ts"
 
 export interface ButtonProps
 	extends ButtonVariants,
 		InteractiveVariants,
-		ComponentProps<"button"> {}
+		ComponentProps<"button"> {
+	color: keyof typeof Colors
```

Comment:
Dont think I was getting auto complete here, so there is probably a better solution to enforce that this is an actual key in the `Colors` record